pkg/alertmanager/alertmanager.go

```diff
@@ -319,12 +328,23 @@ func (am *Alertmanager) Stop() {
 		am.dispatcher.Stop()
 	}
 
+	if service, ok := am.state.(services.Service); ok {
+		service.StopAsync()
+	}
+
 	am.alerts.Close()
 	close(am.stop)
 }
 
 func (am *Alertmanager) StopAndWait() {
 	am.Stop()
+
+	if service, ok := am.state.(services.Service); ok {
+		if err := service.AwaitTerminated(context.Background()); err != nil {
```

Comment:
Shouldn’t we accept a context to `StopAndWait` method, and use that for waiting? (It has “wait” in the name, so it would make sense to me) In such case, we should also return error.